ApollosProject/apollos-apps
# fix: Resolve Daily Habit Timezone Issues

## ✏️ Description
The server needs to know my local timezone in order to accurately tell me if I've completed my Daily Habits with _my_ 24 hour day. This PR updates the front end to make adjustments based on a [backend update](https://github.com/ApollosProject/apollos-apps/pull/2820) that resolves issues resulting in inaccurate UI. Some of the bugs that these PR's resolve include:
- Completing an Experience _today_ marks _yesterday_ as complete
- Dates in the Streak were running from Sunday-Saturday instead of Monday-Sunday
- The id's coming back from the server resulted in inaccurate ID's which made cached results inaccurate

This PR introduces the [`react-native-localize`](https://github.com/zoontek/react-native-localize) package because we need to get the current user's timezone (America/New_York, America/Los_Angeles, Asia/Shanghai) in order to report that to the server and I couldn't find another way to do it.

## 🔬 To Test
1. Pull down the [Apollos Cluster branch](https://github.com/ApollosProject/apollos-apps/pull/2820) that has the new endpoints
2. Run the API locally
3. Start your app and open up Daily Habits
4. Complete 1 or more Experiences to progress through each stage
5. Change the timezone of your computer in order to see how different timezones effect historical data (New York --> Shanghai is a good example)

## ⚠️ To-do Before Merge

- [x] Ensure the [Apollos Cluster PR](https://github.com/ApollosProject/apollos-apps/pull/2820) is merged

## 📝 Checklist
